storage/internal/benchmarks/continuous_reads.go

```diff
@@ -0,0 +1,202 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"math/rand"
+	"os"
+	"path"
+	"slices"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+	"golang.org/x/sync/errgroup"
+)
+
+// continuousReads is a benchmark that continuously downloads the same set of
+// objects from GCS until the timeout is reached.
+//
+// SET UP
+//   - Initialize structs to populate with the benchmark results.
+//   - Select a random API to use to upload and download the object, unless an
+//     API is set in the command-line,
+//   - Select a random size for the objects; this size will
+//     be between two values configured in the command-line.
+//   - Create a directory of objects of that size in GCS, uploading random
+//     contents directly from memory. The number of objects created is equal to
+//     the command-line configuration for the number of objects in a directory.
+//
+// BENCHMARK
+//   - Grab a storage client from the pool.
+//   - Check if the timeout is exceeded; if so, compile p50, p90 and p99 and return.
+//   - Queue a goroutine that performs a download. This goroutine will run as
+//     soon as there is an available worker. The number of workers that run
+//     concurrently is set in the command-line.
+//   - This goroutine chooses the object to download by receiving an object name
+//     from the channel. At the end of the download, it will return that name to
+//     the channel. That way, no goroutines are reading from the same object at
+//     the same time.
```

Comment:
Any particular reason for this? Prince's benchmark is just reading one object; we could just have a round robin of objects to read in a slice without needing to check them out or queue using a channel.

Reply:
In this case the benchmark does still download the objects to tmp. I could change it to io.Discard with a bit of plumbing but I don't think it's strictly necessary - since it overwrites existing objects disk space is not an issue. The channel here is so that threads aren't writing to the same object at once. Do you have a concern about overwriting objects in this manner?

Reply:
Ah okay, I think I just didn't follow the logic completely that existing local files are being overwritten. Seems fine I guess.

Reply:
I added a note in the comment about this